clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -704,7 +704,7 @@ private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionRespons
                             "topic-partition may not exist or the user may not have Describe access to it",
                         batch.topicPartition);
                 } else {
-                    log.warn("Received invalid metadata error in produce request on partition {} due to {}. Going " +
+                    log.warn("Received invalid metadata error in produce request on partition {} due to {} Going " +
                             "to request metadata update now", batch.topicPartition,
```

Comment:
Some error have dot, while others don't.
So, When dot is absent, there is no dot before 'Going'.

See `Errors.SNAPSHOT_NOT_FOUND` for details.

Reply:
Thank you so much for your comment. I have made changes based on your review in the latest (4th) commit.

Reply:
How about including dot in error messages?
I referred to [Use grammatically correct error messages](https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions?redirectedfrom=MSDN#use-grammatically-correct-error-messages).
In addition, most error messages already include dot.

Reply:
This makes a lot of sense. Thank you so much @appchemist. It took me a while to understand the code. I will fix this in my next commit.